### PR TITLE
feat: support tmq

### PR DESCRIPTION
**Pull request by zhaoyanggh**

### CODE REVIEW

File: taos/cinterface.py [hunk 1]
```diff
@@ -951,6 +951,146 @@ def taos_schemaless_insert(connection, lines, protocol, precision):
...
+def tmq_subscription(tmq, topics):
+    res =  _libtaos.tmq_subscription(tmq, byref(topics))
+    if res != 0:
+        raise TmqError(msg="failed on tmq_subscription()", errno=res)
+
+_libtaos.tmq_consumer_poll.argtypes = (c_void_p, c_int64)
+_libtaos.tmq_consumer_poll.restype = c_void_p
+
+def tmq_consumer_poll(tmq, wait_time):
+    return c_void_p(_libtaos.tmq_consumer_poll(tmq, wait_time))
```
Comment (zitsen) [inline taos/cinterface.py:1048]: wrap it to return None or a TaosResult object.

File: taos/tmq.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+from taos.cinterface import *
+from taos.error import *
+from taos.result import *
+
+class TaosTmqConf(object):
```
Comment (zitsen) [inline taos/tmq.py:5]: Add a tmq generator method to conf.

File: taos/tmq.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        tmq_subscribe(self._tmq, list.list())
+        
+    def unsubscribe(self):
+        tmq_unsubscribe(self._tmq)
+        
+    def subscription(self, topics):
+        tmq_subscription(self._tmq, topics.list())
+        
+    def poll(self, time):
+        return tmq_consumer_poll(self._tmq, time)
```
Comment (zitsen) [inline taos/tmq.py:35]: See tmq_consumer_poll comment. Impl the result wraper here or there.

File: taos/tmq.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        
+    def close(self):
```
Comment (zitsen) [inline taos/tmq.py:37]: Add a __del__ method to auto close the tmq.

File: taos/tmq.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        self._list = tmq_list_new()
+        
+    def append(self, topic):
+        tmq_list_append(self._list, topic)
+        
+    def size(self):
+        return tmq_list_get_size(self._list)
+    
+    def destroy(self):
+        tmq_list_destroy(self._list)
```
Comment (zitsen) [inline taos/tmq.py:55]: use `__del__` method to auto destry the tmq_list C object

File: taos/tmq.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        self._conf = tmq_conf_new()
+        
+    def set(self, key, value):
+        tmq_conf_set(self._conf, key, value)
+        
+    def set_auto_commit_cb(self, cb, param):
+        tmq_conf_set_auto_commit_cb(self._conf, cb, param)
+    
+    def destroy(self):
+        tmq_conf_destroy(self._conf)
```
Comment (zitsen) [inline taos/tmq.py:16]: `destroy` should be private, use `__del__` method to auto destroy C object to avoid memory leak.

**Review by zitsen — CHANGES_REQUESTED** [review]
See my comments.

Some suggestions:
1. type annotation for each method
2. write inline doc for publish methods

**Review by zitsen — APPROVED** [review]
LGTM.

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**